src/platform/OpenThread/MdnsImpl.cpp

```diff
@@ -62,25 +62,20 @@ CHIP_ERROR ChipMdnsPublishService(const MdnsService * service)
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
 }
 
-CHIP_ERROR ChipMdnsStopPublish()
+CHIP_ERROR ChipMdnsResetServices()
 {
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
-    return ThreadStackMgr().RemoveAllSrpServices();
+    ThreadStackMgr().InvalidateAllSrpServices();
+    return CHIP_NO_ERROR;
 #else
     return CHIP_ERROR_NOT_IMPLEMENTED;
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
 }
 
-CHIP_ERROR ChipMdnsStopPublishService(const MdnsService * service)
+CHIP_ERROR ChipMdnsCommitServices()
```

Comment:
Just to clarify the flow for SRP, is it:
1. Mark all services as removed (Reset)
2. Add some services again (Advertise)
3. Remove those services that wasn't marked as to add again. (Commit)
?

Why do we call it `CommitServices` method, as actually it doesn't prevent from publishing changes earlier? I know that it will work in most cases, as methods are called quickly and OpenThread will send it as one update, but in fact `Commit` method just removes unused services. Maybe it would be better to call it `FinishServicesUpdate` or `CompleteServicesUpdate` that suggest some post update actions, while `Commit` is like pushing changes?

Reply:
But it's ok to add services one-by-one, so we're not relying on the OT update window. What is problematic is removing and then re-adding a service because we don't want to send an update which removes the service. Correct?

Reply:
I was thinking about `CompleteServiceUpdate`, too. Can rename it.

Reply:
> What is problematic is removing and then re-adding a service because we don't want to send an update which removes the service. Correct?
Right, so changing the name should be enough for me